extensions/wikia/Discussions/utils/DiscussionsVarToggler.class.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+class DiscussionsVarToggler {
+
+	// Discussions related wg variables
+	const ENABLE_DISCUSSIONS_NAV = 'wgEnableDiscussionsNavigation';
+	const ENABLE_DISCUSSIONS = 'wgEnableDiscussions';
+	const ENABLE_FORUMS = 'wgEnableForumExt';
+	const ARCHIVE_WIKI_FORUMS = 'wgArchiveWikiForums';
+
+	private $discussionsVarMap;
+	private $cityId;
+
+	public function __construct( $cityId = null ) {
+		$this->cityId = $cityId ?? F::app()->wg->CityId;
+		$this->discussionsVarMap = [
+			self::ENABLE_DISCUSSIONS => null,
+			self::ENABLE_DISCUSSIONS_NAV => null,
+			self::ENABLE_FORUMS => null,
+			self::ARCHIVE_WIKI_FORUMS => null
+		];
+	}
+
+	public function setEnableDiscussions( bool $val ) : DiscussionsVarToggler {
+		$this->discussionsVarMap[self::ENABLE_DISCUSSIONS] = $val;
+		return $this;
+	}
+
+	public function setEnableDiscussionsNav( bool $val ) : DiscussionsVarToggler {
+		$this->discussionsVarMap[self::ENABLE_DISCUSSIONS_NAV] = $val;
+		return $this;
+	}
+
+	public function setEnableForums( bool $val ) : DiscussionsVarToggler {
+		$this->discussionsVarMap[self::ENABLE_FORUMS] = $val;
+		return $this;
+	}
+
+	public function setArchiveWikiForums( bool $val ) : DiscussionsVarToggler {
+		$this->discussionsVarMap[self::ARCHIVE_WIKI_FORUMS] = $val;
+		return $this;
+	}
```

Comment:
You're combining a builder pattern with a class that actually does other stuff, so you might want to separate the discussionsVarMap into another class which is an argument for the DiscussionsVarToggler.